pkg/controller/petset/pet_set.go

```diff
@@ -222,12 +222,7 @@ func (psc *PetSetController) getPodsForPetSet(ps *apps.PetSet) ([]*api.Pod, erro
 	if err != nil {
 		return []*api.Pod{}, err
 	}
-	// TODO: Do we need to copy?
-	result := make([]*api.Pod, 0, len(pods))
-	for i := range pods {
-		result = append(result, &(*pods[i]))
```

Comment:
actually the intent is to protect the list (i.e if Pods.List() returns a saved list every time, and we just return that, doing returnedList[1]=*Pod{} would reflect as returnedList[1] in the callers), not the elements in the list (that's what the todo is for).